mockafka/aiokafka/aiokafka_consumer.py

```diff
@@ -87,35 +88,39 @@ def __init__(self, *topics: str, **kwargs: Any) -> None:
         self.kafka = KafkaStore()
         self.consumer_store: dict[str, int] = {}
         self.subscribed_topic = [x for x in topics if self.kafka.is_topic_exist(x)]
+        self._is_closed = True
 
     async def start(self) -> None:
         self.consumer_store = {}
+        self._is_closed = False
 
     async def stop(self) -> None:
         self.consumer_store = {}
+        self._is_closed = True
```

Comment:
I'd encourage flipping the order of these statements to avoid races. (Consider what happens if someone manages to call `getone` in the space between these two assignments). While I don't think races are likely here, it's probably good practice to have the ordering the safe way around.

Reply:
Based on the [aiokafka implementation for consumer](https://github.com/aio-libs/aiokafka/blob/master/aiokafka/consumer/consumer.py#L508), in the stop() method, the line self._is_closed = True is used.

However, I'm trying to understand how a race condition can occur in this context. In mockafka, everything works in a single process. Could you explain how race conditions might arise here?

Reply:
look at these lines in `aiokafka`

[L339](https://github.com/aio-libs/aiokafka/blob/master/aiokafka/consumer/consumer.py#L339)
[L1154](https://github.com/aio-libs/aiokafka/blob/master/aiokafka/consumer/consumer.py#L1154)
[L1200](https://github.com/aio-libs/aiokafka/blob/master/aiokafka/consumer/consumer.py#L1200)
[L1261](https://github.com/aio-libs/aiokafka/blob/master/aiokafka/consumer/consumer.py#L1261)

Reply:
> However, I'm trying to understand how a race condition can occur in this context. In mockafka, everything works in a `single process`. Could you explain how race conditions might arise here?

Yeah, this is why I said races were unlikely here. (Nothing stops someone running this code in a threaded context, but it's unlikely).
Having thought about this a bit more, since we're not doing any locking the alternate order of operations doesn't make much difference. Sorry, nevermind.